GitUI/CommandsDialogs/FormDiff.cs

```diff
@@ -280,7 +285,11 @@ private void diffContextToolStripMenuItem_Opening(object sender, System.Componen
 
         private ContextMenuDiffToolInfo GetContextMenuDiffToolInfo()
         {
-            var parentIds = DiffFiles.SelectedItems.Select(i => i.FirstRevision.ObjectId).Distinct().ToList();
+            var parentIds = DiffFiles.SelectedItems
+                .Where(i => i.FirstRevision != null)
+                .Select(i => i.FirstRevision.ObjectId)
+                .Distinct()
```

Comment:
We're executing this selector multiple times. 
Have you considered creating a property on `FileStatusList` (similar to `SelectedGitItems`) to simply the call sites?

E.g. something like:
```cs
    public sealed partial class FileStatusList : GitModuleControl
    {
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        [Browsable(false)]
        public IEnumerable<FileStatusItem> SelectedGitObjects
        {
            get => SelectedItems
                .Where(i => i.FirstRevision != null)
                .Select(i => i.FirstRevision.ObjectId)
                .Distinct();
        }
```

Reply:
It was a little like this before, with many access functions. These properties applies to Enumerate<FileItemStatus> for the selected items. Similar could apply to all items etc,
But I do not know how to do this in the best way, I assumed that a static helper on FileStatusItem would not be accepted, so I let that be.
Similar for First, Second, GitStatus.
(I was not ready thinking about this very late with #7899 and did not want to do any more changes...)

SelectedGitItems should be removed too, but it appeared in a little too many places.

Reply:
I added extension methods, OK now?